src/Data/Shared/RoutesData.php

```
@@ -10,6 +10,7 @@ public function __construct(
         public RouteNameData $names,
         public string $namePrefix,
         public bool $redirect,
+        public ?string $hide,
```

String?

---

Perhaps there was a slight misunderstanding, I meant that we add a new parameter to env, in which we specify a local one, which will be considered the default and if it matches, we hide the parameter from the URL, this is the reason why the line is specified

---

The parameter name is set via the settings file:

https://github.com/Laravel-Lang/config/blob/6efe13fdf4c0a137a559a0067851d8f425be9ac6/config/public.php#L113

Through the environment settings, we must either allow redirects from the default localization to the backup localization or not.

For example:

```
// locale = fr
// fallback = de

// when $hide = true
GET /fr/some >> no redirect
GET /de/some - 302 redirect to GET /some
GET /it/some - no redirect

// when $hide = false
GET /fr/some >> no redirect
GET /de/some - no redirect
GET /it/some - no redirect
```

Because the `app.locale` value can change on the fly and you can't redirect based on it, otherwise you can easily get into recursion.